python/cudf/cudf/io/parquet.py

```diff
@@ -175,29 +174,47 @@ def read_parquet(
 ):
     """{docstring}"""
 
-    filepath_or_buffer, compression = ioutils.get_filepath_or_buffer(
-        filepath_or_buffer, None, **kwargs
-    )
-    if compression is not None:
-        raise ValueError("URL content-encoding decompression is not supported")
+    # Multiple sources are passed as a list. If a single source is passed,
+    # wrap it in a list for unified processing downstream.
+    if not isinstance(filepath_or_buffer, list):
+        filepath_or_buffer = [filepath_or_buffer]
```

Comment:
This is confusing if someone passes say an `tuple` of filenames, because we'll get a list with a single `tuple` in it instead of a list of filenames. I believe we have a utility function that we can use for `is_list_like`: https://github.com/rapidsai/cudf/blob/e3624d0679cb2c383123fca14286049a65708f2a/python/cudf/cudf/utils/dtypes.py#L211-L226

Reply:
thanks for the pointer, useful for both sources and row groups. Fixed the code to sue this.